### ISSUE TITLE
need support literal `Makefile` full name in option `--make`

### ISSUE THREAD

**Issue by yantaozhao** [ggreer/the_silver_searcher#1214]
Please add the full name support for makefile like: `GNUmakefile`, `makefile` and `Makefile` in option `--make` by default.
Because tons of `Makefile` is used as makefile in variety of projects on Linux, etc.

See: https://www.gnu.org/software/make/manual/html_node/Makefile-Names.html

Thanks.

**Comment by aswild** [comment]
I totally agree, which is why I fixed it in my own fork: https://github.com/aswild/the_silver_searcher/commit/12195898eb969ac30221d28cef25d53255a9d807

The fix is a bit non-trivial since the current implementation only can look at file extensions, but it's been working well for me.

I can cherry-pick to master and open a PR here, but not many of those seem to get merged here.

**Comment by trinitronx** [comment]
Just realized after trying to search with `ag --make 'foo'` that it was completely missing all standard `Makefile` within the directory!

Totally misleading that this option doesn't actually work with standard makefile names, and only appears to support `.mk` and `.mak`:

`ag --list-file-types` states it should support `.Makefiles`, which is also a bit confusing and misleading (is it a literal `.Makefiles`, why is it plural?

```
  --make
      .Makefiles  .mk  .mak
```
